### PR TITLE
fix: close mobile nav on blur of nav list

### LINKED ISSUE #379: Keyboard focus going on hidden elements in smaller resolutions
**Describe the bug**
When Navigation items are opened on smaller resolutions and keyboard is used for navigation elements on the page then focus goes to page elements (which are actually hidden) after nav list.

**To Reproduce**
1. Go to https://accessibleweb.dev/ (in mobile view)
2. Click on hamburger menu icon on top right
3. Use keyboard to navigate through focusable elements
4. Focus will move to hidden elements after completing nav list

**Expected behavior**
Focus should not move to hidden elements.

**Recording**
https://github.com/AccessibleForAll/AccessibleWebDev/assets/17885747/7f25aeb7-878f-4b56-8f50-5d6b10c3d4ee

### PR DESCRIPTION
**Pull request by SatyamSetia**
## Describe your changes
This PR contains 2 changes as follows - 
1. Solves the issue #379 
2. Minor refactoring - rename state variables in Layout.tsx and related child components and stories (name changed from showNavMobile to isNavMobileOpen)

## Link to issue
Closes #379 

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] Followed the repository's [Contributing Guidelines](/CONTRIBUTING.md).
- [x] I ran the app and tested it locally to verify that it works as expected.
- [x] I have checked my code with an automatic accessibility tool such as Axe Dev Tools or Wave 
      and it shows no errors.
## Additional Information (Optional)
Any additional information that you want to give us.

### CODE REVIEW

**Comment by SatyamSetia** [comment]
Hey, @EmmaDawsonDev
Requesting your review on this PR whenever you get time.

File: components/Nav/NavPrimaryMobile.tsx [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
+	closeNavMobile: () => void
 }
 
 export const NavPrimaryMobile = ({
 	activeNavLink,
-	handleNavClick,
+	closeNavMobile,
 }: INavProps) => {
+	const handleNavBlur = useCallback((event: FocusEvent<HTMLAnchorElement>) => {
+		const isLastNavItem = /true/i.test(event.target.dataset?.lastNavItem ?? "")
```
Comment (EmmaDawsonDev) [inline components/Nav/NavPrimaryMobile.tsx:16]: I'm not really sure what's going on in this line. I've not seen this syntax before. Can you explain what's happening?

**Review by EmmaDawsonDev — CHANGES_REQUESTED** [review]
Hi Satyam,

I'm not sure all the refactoring/name changing was really necessary. In the future it would be better to discuss these possible changes before going ahead and making them. I prefer to keep pull requests small and focused on one thing. The refactoring should be a separate pull request if it's decided that it's a needed change.

With regards to the navigation bug. It now closes as expected when tabbing forwards. However, if you tab to the last item and then choose to shift+tab to go backwards it also triggers the nav to close and this is not the expected behaviour.

File: components/Nav/NavPrimaryMobile.tsx (hunk 1 above)
Comment (meelix) [reply components/Nav/NavPrimaryMobile.tsx:16]: it's a regex shorthand checking if the tested string contains `true` and returning a boolean. the tested string is either whatever `event.target.dataset?.lastNavItem` returns or an empty string. a needlessly complex way of doing that.

**Pull request closed without merging**